### PR TITLE
fix(AuditLog): Fix log panic while st is nil

### PR DESCRIPTION
**Pull request by MoonJustry**
## Description
Fix st(*logger.ResponseWriter) nil pointer bug

## How to test this PR?
For example, client put request with Accept-Encoding：gzip，the gzhttp.ResponseWriter will overwrite logger.ResponseWriter, then panic.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

File: internal/logger/audit.go [hunk 1]
```diff
@@ -183,13 +184,22 @@ func AuditLog(ctx context.Context, w http.ResponseWriter, r *http.Request, reqCl
...
 
-		st, ok := w.(*ResponseWriter)
-		if ok {
+		var st *ResponseWriter
+		switch v := w.(type) {
+		case *ResponseWriter:
+			st = v
+		case *gzhttp.GzipResponseWriter:
+			// the writer may be obscured by gzip response writer
+			st = v.ResponseWriter.(*ResponseWriter)
```
Comment (klauspost) [inline internal/logger/audit.go]: Lets remove the unchecked assert:
```suggestion
			if rw, ok = v.ResponseWriter.(*ResponseWriter); ok {
				st = rw
			}
```

**Review by klauspost — CHANGES_REQUESTED** [review]
Just one change. Rest LGTM.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by vadmeste — APPROVED** [review]
LGTM, Thanks!

**Review by klauspost — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**